libtsuba/src/RDGPartHeader.h

```diff
@@ -87,6 +92,16 @@ class PropStorageInfo {
 
   const std::string& name() const { return name_; }
   const std::string& path() const { return path_; }
+  const std::shared_ptr<arrow::DataType>& type() const { return type_; }
+
+  // since we don't have type info in the header don't know the
+  // type when this would have been constructed. Allow others to
+  // fix up the type in this case, required until we can get the type
+  // from PartHeader files
+  void set_type(std::shared_ptr<arrow::DataType> type) {
```

Comment:
Ah, this is why you are passing around the type to `WasModified`.  Yeah, this cost is probably worth the price of not changing the on-disk format.  Though I fear changing the on-disk format will only get more difficult...